rest/data/test_results.go

```diff
@@ -1 +1,127 @@
+	"github.com/evergreen-ci/cedar/rest/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/pail"
+	"github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+/////////////////////////////
+// DBConnector Implementation
+/////////////////////////////
+
+func (dbc *DBConnector) FindTestResultByTestName(ctx context.Context, opts TestResultsOptions) (*model.APITestResult, error) {
+	dbOpts := dbModel.TestResultsFindOptions{
+		TaskID:         opts.TaskID,
+		Execution:      opts.Execution,
+		EmptyExecution: opts.EmptyExecution,
+	}
+
+	testResults := &dbModel.TestResults{}
+	testResults.Setup(dbc.env)
+	if err := testResults.FindByTaskID(ctx, dbOpts); db.ResultsNotFound(err) {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("tests with task id '%s' not found", opts.TaskID),
+		}
+	} else if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("%s", errors.Wrap(err, "retrieving metadata")),
+		}
+	}
+
+	bucket, err := testResults.GetBucket(ctx)
+	if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("%s", errors.Wrap(err, "getting bucket")),
+		}
+	}
+
+	return getAPITestResultFromBucket(ctx, bucket, opts.TestName)
+}
+
+///////////////////////////////
+// MockConnector Implementation
+///////////////////////////////
+
+func (mc *MockConnector) FindTestResultByTestName(ctx context.Context, opts TestResultsOptions) (*model.APITestResult, error) {
+	var testResults *dbModel.TestResults
+	for _, tr := range mc.CachedTestResults {
+		if tr.Info.TaskID == opts.TaskID && (opts.EmptyExecution || tr.Info.Execution == opts.Execution) {
+			testResults = &tr
+		}
+	}
+
+	if testResults == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("tests with task id '%s' not found", opts.TaskID),
+		}
+	}
+
+	bucketOpts := pail.LocalOptions{
+		Path:   mc.Bucket,
+		Prefix: testResults.Artifact.Prefix,
+	}
+	bucket, err := pail.NewLocalBucket(bucketOpts)
+	if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("%s", errors.Wrap(err, "problem creating bucket")),
+		}
+	}
+
+	return getAPITestResultFromBucket(ctx, bucket, opts.TestName)
+}
+
+func getAPITestResultFromBucket(ctx context.Context, bucket pail.Bucket, testName string) (*model.APITestResult, error) {
+	tr, err := bucket.Get(ctx, testName)
+	if pail.IsKeyNotFoundError(err) {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("test with specified task id and test name not found"),
+		}
+	} else if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("%s", errors.Wrap(err, "retrieving test result")),
```

Comment:
style nit: you should just be able to use `errors.Wrap(err, "<message>").Error()` to get the stringified version of the error rather than using `fmt.Sprintf`.